### PR TITLE
Feature/restaurant categorie

### PR DESCRIPTION
**Pull request by Nicolas-Rodriguez-Ch**
## Summary

This pull request adds several new features and improvements to the TableTop app. It includes the following changes:

- Refactors functionality to add a facility per venue.
- Adds functionality to add a cuisine to a restaurant.
- Adds an active field to `cuisine_per_restaurant` and soft deletion to it.
- Fixes a minor bug.

## Details

This commit is an update to the TableTop app, as it introduces several new features and improvements. The commit includes a refactor of the functionality to add a facility per venue, which will improve the performance and maintainability of the app.

In addition, the commit adds the ability to add a cuisine to a restaurant, which is an important feature for managing the menu. The commit also adds an active field to `cuisine_per_restaurant` and soft deletion to it, which makes it easier to manage the cuisines associated with each restaurant.

The commit also includes a fix for a minor bug, which will improve the stability of the app.

### CODE REVIEW

File: src/api/restaurantVenues/restaurantVenues.controller.ts [hunk 1]
```diff
@@ -85,3 +86,20 @@ export const deleteRestaurantVenuesController = async (
...
 };
+
+export const updateVenueImageController = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const { id } = req.params;
+    const venue_photo = req.body.venue_photo;
```
Comment (ht1204) [inline src/api/restaurantVenues/restaurantVenues.controller.ts:97]: `const { venue_photo } = req.body;`

File: src/api/users/users.controllers.ts [hunk 2]
```diff
@@ -96,6 +97,25 @@ export const updateUserRoleController = async (req: Request, res: Response) => {
   }
 };
 
+export const removeResAdminController = async (req: Request, res: Response) => {
+  try {
+    const { email, user_role } = req.body;
+    console.log("🚀 ~ file: users.controllers.ts:102 ~ removeResAdminController ~ req.body:", req.body)
+    console.log("🚀 ~ file: users.controllers.ts:102 ~ removeResAdminController ~ user_role:", user_role)
+    console.log("🚀 ~ file: users.controllers.ts:102 ~ removeResAdminController ~ email:", email)
```
Comment (ht1204) [inline src/api/users/users.controllers.ts:105]: are these console.logs necessary yet?

File: src/api/restaurants/restaurants.controller.ts [hunk 3]
```diff
@@ -161,30 +168,46 @@ export const getRestaurantByUserController = async (
...
     await updateUserRole(email, RESTAURANT_ADMIN_ROLE);
 
     const updatedRestaurant = await addAdminToRestaurant(email, restaurantId);
-    res.status(200).json({ message: 'Admin added to the restaurant', data: updatedRestaurant });
+    res
+      .status(200)
+      .json({
+        message: "Admin added to the restaurant",
+        data: updatedRestaurant,
+      });
```
Comment (ht1204) [inline src/api/restaurants/restaurants.controller.ts:207]: why is this not returning the response?

File: src/api/restaurants/restaurants.controller.ts [hunk 4]
```diff
@@ -87,18 +88,22 @@ export const createRestaurantController = async (
...
-    
+
     const existingUser = await getUserByEmail(adminEmail);
     if (!existingUser) {
-      res.status(400).json({ message: "Admin email does not match any existing user" });
+      res
+        .status(400)
+        .json({ message: "Admin email does not match any existing user" });
       return;
     }
```
Comment (ht1204) [inline src/api/restaurants/restaurants.controller.ts:98]: `const { user_id: id } = existingUser;`

File: src/api/restaurantVenues/restaurantVenues.controller.ts [hunk 5]
```diff
@@ -85,3 +86,20 @@ export const deleteRestaurantVenuesController = async (
...
+    const restaurantVenues = await updateVenueImage(id, venue_photo);
+    res
+      .status(200)
+      .json({ message: "Venue image updated", data: restaurantVenues });
```
Comment (ht1204) [inline src/api/restaurantVenues/restaurantVenues.controller.ts:101]: is necessary not returning the response?

File: src/api/restaurantVenues/restaurantVenues.controller.ts (hunk 1 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/api/restaurantVenues/restaurantVenues.controller.ts:97]: Resolved

File: src/api/users/users.controllers.ts (hunk 2 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/api/users/users.controllers.ts:105]: Resolved

File: src/api/restaurants/restaurants.controller.ts (hunk 3 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/api/restaurants/restaurants.controller.ts:207]: resolved

File: src/api/restaurants/restaurants.controller.ts (hunk 4 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/api/restaurants/restaurants.controller.ts:98]: Resolved

File: src/api/restaurantVenues/restaurantVenues.controller.ts (hunk 5 above)
Comment (Nicolas-Rodriguez-Ch) [reply src/api/restaurantVenues/restaurantVenues.controller.ts:101]: Resolved

**Comment by Nicolas-Rodriguez-Ch** [comment]
Changes according to feedback

**Review by ht1204 — APPROVED** [review]

**Pull request merged**